Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -150,4 +151,105 @@ public static int CountSubstring(this string str, string substring, StringCompar
 
         return count;
     }
+
+    /// <summary>
+    /// Determines if the <paramref name="value"/> is longer than 8 characters or contains an <see cref="Environment.NewLine"/>.
+    /// </summary>
+    public static bool IsLongOrMultiline(this string value)
+    {
+        const int humanReadableLength = 8;
+        return value.Length > humanReadableLength || value.Contains(Environment.NewLine, StringComparison.Ordinal);
+    }
+
+    /// <summary>
+    /// Get the mismatch segment between <paramref name="expected"/> and <paramref name="subject"/> for long strings,
+    /// when they differ at index <paramref name="firstIndexOfMismatch"/>.
+    /// </summary>
+    public static string GetMismatchSegmentForLongStrings(this string subject, string expected, int firstIndexOfMismatch)
+    {
+        int trimStart = CalculateSegmentStart(subject, firstIndexOfMismatch);
+
+        int whiteSpaceCount = (firstIndexOfMismatch - trimStart) + 3;
```

Comment:
❓ I don't get what you are trying to do here. Where is the `3` coming from and why do you call this `whiteSpaceCount`. I'm pretty sure it contains other characters as well.

Reply:
It's the number of visible white-space before the arrow symbol. The `3` comes from the length of the prefix `  "` that's before every text and the optional increment considers the ellipsis that is included.
I renamed it to `whiteSpaceCountBeforeArrow` and replaced the `3` with `prefix.Length`.